xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java

```diff
@@ -69,7 +69,7 @@
   static final long RPC_TIMEOUT_SECONDS = 10L;
 
   private static final Pattern CLUSTER_URL_PATTERN = Pattern
-      .compile(".*/projects/(.*)/locations/(.*)/clusters/.*");
+      .compile(".*/projects/(.*)/(?:locations|zones)/(.*)/clusters/.*");
```

Comment:
Is `(.*)` the same as `.*`? Seems you are not capturing groups.

Reply:
`(.*)` is a capturing group and I am capturing only 2 groups (to get the project name and the cluster name). The only change here is from:

`locations`

to

`(?:locations|zones)`

And the initial `?:` makes it a [non-capturing group](https://stackoverflow.com/questions/3512471/what-is-a-non-capturing-group-in-regular-expressions).

Reply:
> to get the project name and the cluster name

To get the cluster name, why not `/clusters/(.*)`? I might misunderstand the URI.

Reply:
Oh, I see you are actually getting project name and zone name 
https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/internal/certprovider/MeshCaCertificateProviderProvider.java#L245-L246

Reply:
Oops, sorry I made a mistake in my previous comment. I don't need the cluster name but just the project name and the location (or zone) name. And that's why use the first capturing group for the part following `/projects/` and the second one for the part following either `/locations/` or `/zones/` which is why I have the `(?:locations|zones)` expression.

The GkeClusterUrl is of the form: `https://container.googleapis.com/$VERSION/projects/$PROJECT/$LOCATIONS/$ZONE/clusters/$CLUSTER`

where $LOCATIONS is either `locations` or `zones`